From a1696741375c6faca0d09ae6b271a2c56fc2b6fe Mon Sep 17 00:00:00 2001
From: Dengke Du <dengke.du@windriver.com>
Date: Tue, 7 May 2019 15:26:32 +0800
Subject: [PATCH] tools: add libvirt-net-rpc to virt-host-validate when TLS is
 enabled

When gnu-tls is enabled for libvirt references to virNetTLSInit are
generated in libvirt. Any binaries linking against libvirt, must also
link against libvirt-net-rpc which provides the implementation.

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
[ywei: rebased to libvirt-1.3.2]
Signed-off-by: Yunguo Wei <yunguo.wei@windriver.com>
[MA: rebase to v4.3.0]
Signed-off-by: Mark Asselstine <mark.asselstine@windriver.com>
[ddu: rebase to v5.3.0]
Signed-off-by: Dengke Du <dengke.du@windriver.com>

---
 examples/Makefile.am | 20 ++++++++++++++++++++
 tools/Makefile.am    | 12 ++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/examples/Makefile.am b/examples/Makefile.am
index ad635bd..a94f41d 100644
--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -74,6 +74,10 @@ LDADD = \
 	$(top_builddir)/src/libvirt-admin.la \
 	$(NULL)
 
+if WITH_GNUTLS
+LDADD += $(top_builddir)/src/libvirt-net-rpc.la
+endif
+
 noinst_PROGRAMS = \
 	c/admin/client_close \
 	c/admin/client_info \
@@ -111,6 +115,22 @@ c_misc_openauth_SOURCES = c/misc/openauth.c
 examplesdir = $(docdir)/examples
 
 adminexamplesdir = $(examplesdir)/c/admin
+
+if WITH_GNUTLS
+dominfo_info1_LDADD = $(top_builddir)/src/libvirt-net-rpc.la  \
+               $(LDADD)    \
+               $(NULL)
+domsuspend_suspend_LDADD = $(top_builddir)/src/libvirt-net-rpc.la  \
+               $(LDADD)    \
+               $(NULL)
+hellolibvirt_hellolibvirt_LDADD = $(top_builddir)/src/libvirt-net-rpc.la  \
+               $(LDADD)    \
+               $(NULL)
+openauth_openauth_LDADD = $(top_builddir)/src/libvirt-net-rpc.la  \
+               $(LDADD)   \
+               $(NULL)
+endif
+
 adminexamples_DATA = $(ADMIN_EXAMPLES)
 
 domainexamplesdir = $(examplesdir)/c/domain
diff --git a/tools/Makefile.am b/tools/Makefile.am
index 53df930..2a0a989 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -166,6 +166,12 @@ virt_host_validate_LDADD = \
 		$(GLIB_LIBS) \
 		$(NULL)
 
+if WITH_GNUTLS
+virt_host_validate_LDADD += ../src/libvirt-net-rpc.la \
+                            ../gnulib/lib/libgnu.la   \
+                            $(NULL)
+endif
+
 virt_host_validate_CFLAGS = \
 		$(AM_CFLAGS) \
 		$(NULL)
@@ -262,6 +268,12 @@ virt_admin_CFLAGS = \
 		$(READLINE_CFLAGS)
 BUILT_SOURCES =
 
+if WITH_GNUTLS
+virsh_LDADD += ../src/libvirt-net-rpc.la \
+               ../gnulib/lib/libgnu.la   \
+               $(NULL)
+endif
+
 if WITH_WIN_ICON
 virsh_LDADD += virsh_win_icon.$(OBJEXT)
 
